### ISSUE TITLE
单机多卡跑grpo，多个step后会报错

### ISSUE THREAD

**Issue by shunxing12345** [modelscope/ms-swift#3576]
**Describe the bug**
单机多卡跑grpo，多个step后会报错
`
MAX_PIXELS=1003520 
CUDA_VISIBLE_DEVICES=0,1,2,3 
NPROC_PER_NODE=4 
swift rlhf \
--rlhf_type grpo \
--model /gemini-1/space/space/private/xiangwen/telechat_model/TeleChat2-3B \
--train_type lora \
--dataset /gemini-1/space/space/private/xiangwen/public_proj/AI-MO__numina_math-tir \
--torch_dtype float16 \
--system /gemini-1/space/space/private/xiangwen/public_proj/ms-swift/examples/train/grpo/prompt.txt \
--num_train_epochs 1 \
--max_length 2048 \
--per_device_train_batch_size 1 \
--per_device_eval_batch_size 1 \
--learning_rate 1e-6 \
--save_total_limit 2 \
--logging_steps 5 \
--output_dir output \
--gradient_accumulation_steps 1 \
--warmup_ratio 0.05 \
--dataloader_num_workers 2 \
--max_completion_length 1024 \
--reward_funcs accuracy format \
--num_generations 4 \
--use_vllm true \
--vllm_gpu_memory_utilization 0.5 \ 
--sleep_level 1 \
--deepspeed zero2 \
--num_infer_workers 4 \
--tensor_parallel_size 2 \
--temperature 1.0 \
--top_p 0.85

`

<img width="950" alt="Image" src="https://github.com/user-attachments/assets/e13eea9d-b1d8-4a6a-a900-35aeea905777" />

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
torch 2.5.1
GPU A800 * 4
system Linux
CUDA 12.1

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by hjh0119** [comment]
Feel free to reopen if you have any issues.

**Comment by shunxing12345** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
